### PR TITLE
Add: callback for task pool

### PR DESCRIPTION
**Pull request by zouyx**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: sync/task_pool.go [hunk 1]
```diff
@@ -84,14 +84,21 @@ func WithTaskPoolTaskQueueNumber(number int) TaskPoolOption {
 /////////////////////////////////////////
 
 // task t
-type task func()
+type runnable func() (interface{}, error)
```
Comment (wongoo) [inline sync/task_pool.go]: the runnable is not compatible with previous task definition

File: sync/task_pool.go [hunk 2]
```diff
@@ -84,14 +84,21 @@ func WithTaskPoolTaskQueueNumber(number int) TaskPoolOption {
...
+
+type consum func(interface{}, error)
+
+type task struct {
+	run          runnable
```
Comment (wongoo) [inline sync/task_pool.go]: you can define a function as a task , in which calling runnable first and callback the result:
```
pool.AddTask(func(){
    callback(runnable())
})
```

File: sync/task_pool.go (hunk 1 above)
Comment (zouyx) [reply sync/task_pool.go]: yes , i see, once update. i will fix it  in dubbogo

File: sync/task_pool.go (hunk 2 above)
Comment (zouyx) [reply sync/task_pool.go]: good idea.

File: sync/task_pool.go (hunk 1 above)
Comment (zouyx) [reply sync/task_pool.go]: fix it

**Comment by wongoo** [comment]
this pr is accomplished，but i think it's not a common request for task pool. 
some other user may have different definition of  runnable and callback.
what's your opinion?  @AlexStocks @fangyincheng @hxmhlt

File: sync/task_pool.go [hunk 3]
```diff
@@ -160,6 +164,10 @@ func (p *TaskPool) run(id int, q chan task) error {
 
 // add task
 func (p *TaskPool) AddTask(t task) {
+	if t == nil {
+		return
```
Comment (AlexStocks) [inline sync/task_pool.go]: maybe there should be a panic when the @t  nil? @wongoo

File: sync/task_pool_test.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+ */
+
+package gxsync
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
```
Comment (AlexStocks) [inline sync/task_pool_test.go]: another import block, pls.

File: sync/task_pool.go (hunk 3 above)
Comment (wongoo) [reply sync/task_pool.go]: agree

**Comment by wongoo** [comment]
@AlexStocks  do u think the definition of runnable and callback is a common requirement for task pool.  My answer is no. I think it should be a part of special service.

**Pull request closed without merging**